syntax = "proto3";

service MeshBuilder {
    rpc CreateProject(CreateProjectRequest) returns (CreateProjectResponse) {}
    rpc Add(AddRequest) returns (AddResponse) {}
    rpc Retrieve(RetrieveRequest) returns (RetrieveResponse) {}
}

message CreateProjectRequest {
    string name = 1;
}
message CreateProjectResponse { }

message AddRequest {
    string name = 1;
    Depth depth = 2;
}
message AddResponse { }

message RetrieveRequest {
    string name = 1;
}
message RetrieveResponse {
    repeated Point points = 1;
}
message Point {
    float X = 1;
    float Y = 2;
    float Z = 3;
}

message Depth {
    repeated Row rows = 1;

    // FOV in degrees
    float x_fov = 2;
    float y_fov = 3;
}
message Row {
    repeated int32 values = 1;
}
